code/modules/spells/spell_projectile.dm

```diff
@@ -39,11 +39,18 @@
 	return
 
 /obj/item/projectile/spell_projectile/Bump(var/atom/A)
+	if(!carried)
```

Comment:
If the gun fired a spell like fireball or force bolt, this part would runtime due to there being no carrier for the spell.
Adding this check fixed that.